applications/nrf_desktop/src/modules/hfclk_lock.c

```diff
@@ -17,38 +17,39 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(MODULE);
 
-#define DT_DRV_COMPAT nordic_nrf_clock
-
-static struct device *clk_dev;
-
+static struct onoff_manager *mgr;
+static struct onoff_client cli;
 
 static void hfclk_lock(void)
 {
-	if (clk_dev) {
+	if (mgr) {
 		return;
 	}
 
-	clk_dev = device_get_binding(DT_INST_LABEL(0));
-
-	if (!clk_dev) {
+	mgr = z_nrf_clock_control_get_onoff(CLOCK_CONTROL_NRF_SUBSYS_HF);
+	if (!mgr) {
 		module_set_state(MODULE_STATE_ERROR);
 	} else {
-		clock_control_on(clk_dev, CLOCK_CONTROL_NRF_SUBSYS_HF);
-		module_set_state(MODULE_STATE_READY);
+		int err;
+
+		sys_notify_init_spinwait(&cli.notify);
+		err = onoff_request(mgr, &cli);
+		module_set_state((err < 0) ?
+				 MODULE_STATE_ERROR : MODULE_STATE_READY);
```

Comment:
Shouldn't `mgr` be set to `NULL` in case of an error here? In the current form, no further attempt to request HFCLK will be possible.